### ISSUE TITLE
Incorrect IAmqpConnectionManager interface declaration

### ISSUE THREAD

**Issue by radislav** [jwalton/node-amqp-connection-manager#184]
Can't build TS project uses `amqp-connection-manager` because of errors:
```
node_modules/amqp-connection-manager/dist/esm/AmqpConnectionManager.d.ts(103,5): error TS1131: Property or signature expected.
node_modules/amqp-connection-manager/dist/esm/AmqpConnectionManager.d.ts(103,9): error TS1005: ';' expected.
node_modules/amqp-connection-manager/dist/esm/AmqpConnectionManager.d.ts(103,21): error TS1005: ';' expected.
node_modules/amqp-connection-manager/dist/esm/AmqpConnectionManager.d.ts(105,9): error TS1005: ';' expected.
node_modules/amqp-connection-manager/dist/esm/AmqpConnectionManager.d.ts(105,23): error TS1005: ';' expected.
node_modules/amqp-connection-manager/dist/esm/AmqpConnectionManager.d.ts(106,1): error TS1128: Declaration or statement expected.
```

This issue can be resolved by replacing [these lines](https://github.com/jwalton/node-amqp-connection-manager/blob/master/src/AmqpConnectionManager.ts#L115-L119)
```ts
    /** The current connection. */
    get connection(): Connection | undefined;

    /** Returns the number of registered channels. */
    get channelCount(): number;
```

with proper "getter-only" definition
```ts
    /** The current connection. */
    readonly connection: Connection | undefined;

    /** Returns the number of registered channels. */
    readonly channelCount: number;
```

**Comment by radislav** [comment]
Ok. The issue was in TS lib version, but it would be good to support other versions and update the original definition.

**Comment by Dezzley** [comment]
Thanks, @radislav.
I experienced the same problem with version `3.6.0` of `amqp-connection-manager`. Took me a while to figure out that it depends on typescript `^4.3.5` and I had to bump the version in my project to make it work (from `4.1.5`).
It could be tricky for those who use the `nestjs` module [@golevelup/nestjs-rabbitmq](https://github.com/golevelup/nestjs/blob/master/packages/rabbitmq/package.json) as it implicitly requires the updated typescript.
